rules/aip0131/method_signature.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0131
+
+import (
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var methodSignature = &lint.MethodRule{
+	Name:   lint.NewRuleName(131, "method-signature"),
+	OnlyIf: isGetMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		signatures := utils.GetMethodSignatures(m)
+
+		// Check if the signature is missing.
+		if len(signatures) == 0 {
+			return []lint.Problem{{
+				Message: fmt.Sprintf(
+					"Get methods should include `(google.api.method_signature) = %q`",
+					"name",
+				),
+				Descriptor: m,
+			}}
+		}
+
+		// Check if the signature is wrong.
+		if diff := cmp.Diff(signatures[0], []string{"name"}); diff != "" {
```

Comment:
Should we just write `len(signature[0]) == !1 && signature[0][0] == "name"`? Using `cmp` seems to be heavy here.

Reply:
I think `cmp.Diff` is the standard/preferred way to do this sort of thing.

Reply:
I would prefer `cmp` in testing. But using it probably is overkilled here.

Reply:
Okay, using `reflect.DeepEqual` instead.